api/nodes/permissions.py

```diff
@@ -50,6 +50,20 @@ def has_object_permission(self, request, view, obj):
         return obj.is_public or obj.can_view(auth)
 
 
+class IsAdminAndContributor(permissions.BasePermission):
+    acceptable_models = (AbstractNode, DraftRegistration)
+
+    def has_object_permission(self, request, view, obj):
+        assert_resource_type(obj, self.acceptable_models)
+        if isinstance(obj, DraftRegistration):
+            obj = obj.branched_from
+        auth = get_user_auth(request)
+        if request.method in permissions.SAFE_METHODS:
+            return obj.has_permission(auth.user, osf_permissions.ADMIN)
```

Comment:
There were existing tests that checked if a group admin could view drafts, so added this so that this permission would restrict updates but allow views from admins.